tests/_server/api/endpoints/test_resume_session.py

```diff
@@ -0,0 +1,252 @@
+                    },
+                ],
+            },
+        )
+
+    # Check the session still exists after closing the websocket
+    assert get_session(client, "123")
+
+    # New session with new ID (simulates refresh)
+    # We should resume the current session
+    with client.websocket_connect("/ws?session_id=456") as websocket:
+        # First message is the kernel reconnected
+        data = websocket.receive_json()
+        assert data == {"op": "reconnected", "data": {}}
+        # Resume the session
+        data = websocket.receive_json()
+        assert_kernel_ready_response(
+            data,
+            create_response(
+                {
+                    # The cell IDs that were saved should be the ones that are
+                    # resumed
+                    "cell_ids": ["2", "1"],
+                    "names": ["cell_0", "cell_1"],
+                    "codes": [
+                        "slider = mo.ui.slider(0, 100)",
+                        "import marimo as mo",
+                    ],
+                    "configs": [
+                        {
+                            "hideCode": True,
+                            "disabled": True,
+                        },
+                        {
+                            "hideCode": False,
+                            "disabled": False,
+                        },
+                    ],
+                    "resumed": True,
+                }
+            ),
+        )
+
+    # Check the session switch IDs
+    assert not get_session(client, "123")
+    assert get_session(client, "456")
+
+    # Shutdown the kernel
+    client.post("/api/kernel/shutdown", headers=HEADERS)
+
+
+def test_save_config(client: TestClient) -> None:
+    with client.websocket_connect("/ws?session_id=123") as websocket:
+        data = websocket.receive_json()
+        assert_kernel_ready_response(data, create_response({}))
+        # Send save request
+        client.post(
+            "/api/kernel/save_app_config",
+            headers=headers("123"),
+            json={
+                "config": {"width": "full"},
+            },
+        )
+
+    # Check the session still exists after closing the websocket
+    assert get_session(client, "123").app.config.width == "full"
+    assert get_session_manager(client).app_config.width == "full"
+
+    # Loading index page should have the new config
+    response = client.get("/")
+    assert response.status_code == 200
+    assert '"width": "full"' in response.text
+
+    # Shutdown the kernel
+    client.post("/api/kernel/shutdown", headers=HEADERS)
+
+
+def test_restart_session(client: TestClient) -> None:
+    with client.websocket_connect("/ws?session_id=123") as websocket:
+        data = websocket.receive_json()
+        assert_kernel_ready_response(data, create_response({}))
+        # Send save request
+        client.post(
+            "/api/kernel/restart_session",
+            headers=headers("123"),
+        )
```

Comment:
Moving this out of the `with` block fixes one of the failing `Windows` tests. Does that work or does that break the logic of the test?

Reply:
shouldnt effect the logic, with resumed sessions. moving it out